source/slang/slang-ir-specialize.cpp

```diff
@@ -1787,7 +1787,25 @@ struct SpecializationContext
                 // created.
                 //
                 auto valType = val->getFullType();
-                if (auto extractExistentialType = as<IRExtractExistentialType>(valType))
+                if (!isCompileTimeConstantType(valType))
+                {
+                    // If the `argument` is not specialized yet, don't aggressively specialize the
+                    // parameter.
+                    //
+                    // If we specialize the parameter type too early, we will lose the opportunity
+                    // to specialize the callee later. The principal is to always let the
+                    // specialization happen at the same time for both on argument and parameter.
+                    //
+                    // Note we should not use `createParam` here, because this call won't assign the
+                    // parent to the new parameter, therefore during the cloning process, some
+                    // existential related IR inst could be hoisted to the global scope, which is
+                    // unexpected. Instead, we should use cloneInst here, such that the new
+                    // parameter will be inserted into the function scope.
+                    auto newParam = (IRParam*)cloneInst(&cloneEnv, builder, oldParam);
+                    newParams.add(newParam);
+                    continue;
+                }
+                else if (auto extractExistentialType = as<IRExtractExistentialType>(valType))
```

Comment:
This branch will never hit with the new if branch? Because is `valType` is a compiletime constant type, then it cannot be a extract existential type.